cmd/d2/main.go

```diff
@@ -211,9 +212,26 @@ func compile(ctx context.Context, ms *xmain.State, plugin d2plugin.Plugin, theme
 	if err != nil {
 		return nil, err
 	}
+	svg, err = imgbundler.InlineLocal(ctx, ms, svg)
+	if err != nil {
+		if !isWatching {
+			ms.Log.Error.Printf("missing/broken remote image(s), writing partial output: %v", err)
+		} else {
+			ms.Log.Debug.Printf("ignoring missing/broken local image(s) in watch mode: %v", err)
+		}
+	}
 
 	out := svg
 	if filepath.Ext(outputPath) == ".png" {
+		svg, err = imgbundler.InlineRemote(ctx, ms, svg)
+		if err != nil {
+			if !isWatching {
+				ms.Log.Error.Printf("missing/broken remote image(s), writing partial output: %v", err)
+			} else {
+				ms.Log.Debug.Printf("ignoring missing/broken remote image(s) in watch mode: %v", err)
```

Comment:
This should be an Error too. In fact, I don't think the message needs to change at all anymore between watch/nowatch, just always log the error.